src/KOKKOS/atom_vec_spin_kokkos.cpp

```diff
@@ -637,11 +637,12 @@ int AtomVecSpinKokkos::unpack_border_hybrid(int n, int first, double *buf)
 
   m = 0;
   last = first + n;
-  for (i = first; i < last; i++)
+  for (i = first; i < last; i++) {
```

Comment:
@julient31 FYI this was flagged by compiling with clang and using full compiler warnings (-Wall -Wextra)

Reply:
Hey @akohlmey !

Ouch, yes, that was quite horrendous... How did I miss that? ^^

Thanks for finding this Axel!

Reply:
I didn't find it, the Clang compiler did. When compiling with GCC I use `-Wall -Wextra -Wno-maybe-uninitialized` and when using Clang I use `-Wall -Wextra` and that produces warnings about unused variables, set but unused variables, unused local functions, local variables shadowing class members, and so on. When there are warnings about "inconsistent indentation" it is often (but not always) a sign of missing braces like in this case. We try to reduce such "easily fixable" compiler warnings to the minimum (sometimes they cannot be avoided or the fix would only help hiding but not fixing problems) so that new warnings stand out more. This is essentially a first round of static code analysis to help maintaining code quality. As second round it running specific static analysis tools like ["Coverity Scan"](https://scan.coverity.com/projects/lammps) or ["CodeQL"](https://github.com/lammps/lammps/security/code-scanning) that are run either after every merge to the master branch (CodeQL, faster but less capable) or once a week (Coverity Scan, slower but more powerful) and review the reported issues (there are plenty of false positives because of the coding style and flow of control employed in LAMMPS), but they can point out common programming errors that are sometimes hard to find.

Reply:
I took notes and saved your message. Actually, I remember you already mentioned the -Wall -Wextra -Wno-maybe-uninitialized  flags for gcc. I need to more systematically check this before submitting PRs.

Reply:
Thanks a lot for the explanation Axel!

Reply:
> I took notes and saved your message. Actually, I remember you already mentioned the -Wall -Wextra -Wno-maybe-uninitialized flags for gcc. I need to more systematically check this before submitting PRs.

Then you should also make a note to do "make check-whitespace" before submitting and possibly "make fix-whitespace" (and then check and commit the changes for "your" files) to remove whitespace-only changes that produce extra differences when comparing different commits.

Reply:
Done! ;-)